File: tools/java/org.hl7.fhir.instance/src/org/hl7/fhir/instance/model/Binary.java
package org.hl7.fhir.instance.model;

/**
 * A class that represents the special FHIR resource "Binary"
 */
public class Binary extends Resource {

  private static final long serialVersionUID = -5374982878067063027L;

	@Override
  public ResourceType getResourceType() {
    return ResourceType.Binary;
  }

  /**
   * The content of the binary
   */
  private byte[] content;
  
  /**
   * The mimetype of the binary content
   */
  private String contentType;
  
  /**
   * @return The content of the binary
   */
  public byte[] getContent() {
    return content;
  }
  /**
   * @param content The content of the binary
   */
  public void setContent(byte[] content) {
    this.content = content;
  }
  /**
   * @return The mimetype of the binary content
   */
  public String getContentType() {
    return contentType;
  }
  /**
   * @param contentType The mimetype of the binary content
   */
  public void setContentType(String contentType) {
    this.contentType = contentType;
  }
  
  
}

File: implementations/csharp/Model.Support/Binary.cs
/*
  Copyright (c) 2011-2012, HL7, Inc
  All rights reserved.
  
  Redistribution and use in source and binary forms, with or without modification, 
  are permitted provided that the following conditions are met:
  
   * Redistributions of source code must retain the above copyright notice, this 
     list of conditions and the following disclaimer.
   * Redistributions in binary form must reproduce the above copyright notice, 
     this list of conditions and the following disclaimer in the documentation 
     and/or other materials provided with the distribution.
   * Neither the name of HL7 nor the names of its contributors may be used to 
     endorse or promote products derived from this software without specific 
     prior written permission.
  
  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND 
  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED 
  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. 
  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, 
  INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT 
  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR 
  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
  POSSIBILITY OF SUCH DAMAGE.
  

*/

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Hl7.Fhir.Model
{
    // A stream of bytes
    public partial class Binary : Resource
    {
        public string ContentType { get; set; }
        
        public byte[] Content { get; set; }

        public static bool TryParse(string value, out Binary result)
        {
            try
            {
                byte[] binData = null;

                if (value == null)
                    result = null;
                else
                    binData = Convert.FromBase64String(value);

                result = new Binary(binData);
                return true;
            }
            catch
            {
                result = null;
                return false;
            }
        }

        public Binary(byte[] data, string contentType=null)
        {
            this.Content = data;
            this.ContentType = contentType;
        }

        public static Binary Parse(string value)
        {
            Binary result = null;

            if (TryParse(value, out result))
                return result;
            else 
                throw new FhirFormatException("Not a correctly base64 encoded binary value");
        }


        public override string ToString()
        {
            if (Content == null)
                return null;
            else
                return Convert.ToBase64String(Content);
        }
    }
  
}
